test/core/end2end/end2end_tests.h

```diff
@@ -586,140 +607,133 @@ class CoreEnd2endTest : public ::testing::Test {
-    GTEST_SKIP() << "Disabled for Local TCP Connection";            \
+#define SKIP_IF_LOCAL_TCP_CREDS()                                      \
+  if (test_config()->feature_mask & FEATURE_MASK_IS_LOCAL_TCP_CREDS) { \
+    GTEST_SKIP() << "Disabled for Local TCP Connection";               \
   }
 
-#define CORE_END2END_TEST(suite, name)                                       \
-  class CoreEnd2endTest_##suite##_##name : public grpc_core::suite {         \
-   public:                                                                   \
-    CoreEnd2endTest_##suite##_##name() {}                                    \
-    void TestBody() override {                                               \
-      if ((GetParam()->feature_mask & FEATURE_MASK_IS_CALL_V3) &&            \
-          (grpc_core::ConfigVars::Get().PollStrategy() == "poll")) {         \
-        GTEST_SKIP() << "call-v3 not supported with poll poller";            \
-      }                                                                      \
-      RunTest();                                                             \
-    }                                                                        \
-    void RunTest() override;                                                 \
-                                                                             \
-   private:                                                                  \
-    static grpc_core::CoreEnd2endTest* Run(                                  \
-        const grpc_core::CoreTestConfiguration* config) {                    \
-      auto* test = new CoreEnd2endTest_##suite##_##name;                     \
-      test->TestInfrastructureSetParam(config);                              \
-      return test;                                                           \
-    }                                                                        \
-    static int registered_;                                                  \
-  };                                                                         \
-  int CoreEnd2endTest_##suite##_##name::registered_ =                        \
-      (grpc_core::CoreEnd2endTestRegistry::Get().RegisterTest(#suite, #name, \
-                                                              &Run),         \
-       0);                                                                   \
+#ifndef GRPC_END2END_TEST_INCLUDE_FUZZER
+#define CORE_END2END_FUZZER(suite, name)
+#else
+#define CORE_END2END_FUZZER(suite, name)                                \
+  FUZZ_TEST(Fuzzers, suite##_##name)                                    \
+      .WithDomains(::fuzztest::ElementOf(suite::AllSuiteConfigs(true)), \
+                   ::fuzztest::Arbitrary<core_end2end_test_fuzzer::Msg>());
+#endif
+
+// NOLINTBEGIN(bugprone-macro-parentheses)
+#if defined(FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION) || \
+    defined(GRPC_END2END_TEST_NO_GTEST)
+#define CORE_END2END_TEST_P(suite, name)
```

Comment:
I assume the long-term goal here is to get rid of all of these macros and just use the normal gtest parameterized test suites directly?

Reply:
Yeah. And I can't wait.